src/fsharp/FSharp.Core/control.fs

```diff
@@ -1462,6 +1462,50 @@ namespace Microsoft.FSharp.Control
                             |> unfake);
                     FakeUnit))
 
+        static member Choice(computations : Async<'T option> seq) : Async<'T option> =
+            unprotectedPrimitive(fun args ->
+                let result =
+                    try Choice1Of2 <| Seq.toArray computations
+                    with exn -> Choice2Of2 <| ExceptionDispatchInfo.RestoreOrCapture(exn)
+
+                match result with
+                | Choice2Of2 edi -> args.aux.econt edi
+                | Choice1Of2 [||] -> args.cont None
+                | Choice1Of2 [|P body|] -> body args
```

Comment:
This line reflects the original fssnip implementation: if given a singleton array as argument, simply pass your current continuation to it. This does not happen in `Parallel`, so wondering whether this could violate expectations. @dsyme?